adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -455,18 +457,24 @@ private AuthenticationResult postMessage(String requestMessage, Map<String, Stri
             HttpWebResponse response = mWebRequestHandler.sendPost(authority, headers,
                     requestMessage.getBytes(AuthenticationConstants.ENCODING_UTF8),
                     "application/x-www-form-urlencoded");
+            final Map<String, List<String>> responseHeaders = response.getResponseHeaders();
+
+            if (null != responseHeaders && null != responseHeaders.get(X_MS_CLITELEM)) {
+                httpEvent.setSpeRingInfo(responseHeaders.get(X_MS_CLITELEM).get(0));
```

Comment:
Just curious, is it guaranteed that responseHeaders.get(X_MS_CLITELEM) will have at least 1 element?

Reply:
I believe this dereferencing should be safe, but in the event that the header is supplied without a value, I'll add some defensive programming around this 👍